eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -815,12 +821,18 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
 
     case Event(BroadcastChannelUpdate(reason), d: DATA_NORMAL) =>
       val age = TimestampSecond.now() - d.channelUpdate.timestamp
-      val channelUpdate1 = Announcements.makeChannelUpdate(nodeParams.chainHash, nodeParams.privateKey, remoteNodeId, scidForChannelUpdate(d), d.channelUpdate.cltvExpiryDelta, d.channelUpdate.htlcMinimumMsat, d.channelUpdate.feeBaseMsat, d.channelUpdate.feeProportionalMillionths, d.commitments.params.maxHtlcAmount, isPrivate = !d.commitments.announceChannel, enable = Helpers.aboveReserve(d.commitments))
+      val channelUpdate1 = Helpers.makeChannelUpdate(nodeParams, remoteNodeId, scidForChannelUpdate(d), d.commitments, d.channelUpdate.relayFees)
       reason match {
         case Reconnected if d.commitments.announceChannel && Announcements.areSame(channelUpdate1, d.channelUpdate) && age < REFRESH_CHANNEL_UPDATE_INTERVAL =>
           // we already sent an identical channel_update not long ago (flapping protection in case we keep being disconnected/reconnected)
           log.debug("not sending a new identical channel_update, current one was created {} days ago", age.toDays)
           stay()
+        case BalanceThresholdCrossed if d.channelUpdate.htlcMaximumMsat == channelUpdate1.htlcMaximumMsat =>
+          // The current channel update already uses the desired htlcMaximumMsat.
+          stay()
+        case BalanceThresholdCrossed if age < nodeParams.channelConf.minTimeBetweenUpdates =>
+          startSingleTimer(BalanceThresholdCrossed.toString, BroadcastChannelUpdate(BalanceThresholdCrossed), nodeParams.channelConf.minTimeBetweenUpdates - age)
+          stay()
         case _ =>
           log.debug("refreshing channel_update announcement (reason={})", reason)
```

Comment:
How will we monitor/troubleshoot the updates to the `max-htlc-amount` field? Shouldn't we log at the `info` level whenever we change our `max-htlc-amount`? This will create at most one log line per channel per hour, so it shouldn't spam us (compared to many other logs) and would be quite useful to see if we are frequently missing out on fee revenue.